java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java

```diff
@@ -297,7 +297,7 @@ public void close() {
     }
 
     @CheckForNull
-    private <RequestT> OpenSearchRequestBodyBuffer prepareRequestBody(
+    public <RequestT> OpenSearchRequestBodyBuffer prepareRequestBody(
```

Comment:
Those should probably not be public

Reply:
I had to do that since I had to write a test for the header :(

Reply:
I think we need to find a way. How about making it protected and subclassing AwsSdk2Transport?

Reply:
Do you mean something like this?

```
public class AwsSdk2TransportMock extends AwsSdk2Transport {

    public AwsSdk2TransportMock(
            @Nonnull SdkHttpClient httpClient,
            @Nonnull String host,
            @Nonnull String signingServiceName,
            @Nonnull Region signingRegion,
            @CheckForNull AwsSdk2TransportOptions options) {
        super(httpClient, null, host, signingServiceName, signingRegion, options);
    }

    @Override
    public <RequestT> OpenSearchRequestBodyBuffer prepareRequestBody(RequestT request,
            Endpoint<RequestT, ?, ?> endpoint, TransportOptions options) throws IOException {
        return super.prepareRequestBody(request, endpoint, options);
    }
    @Override
    public <RequestT> SdkHttpFullRequest prepareRequest(RequestT request, Endpoint<RequestT, ?, ?> endpoint,
            TransportOptions options, OpenSearchRequestBodyBuffer body) {
        return super.prepareRequest(request, endpoint, options, body);
    }
}
```
where the `prepareRequest` and `prepareRequestBody` in AwsSdk2Transport are `protected`?

Reply:
Right. Looking at your implementation I think that was not a good suggestion, sorry. I say we open a bug for this and call it a day instead of adding the child class.